tests/integration/test_agentd/test_agentd_state/test_agentd_state.py

```diff
@@ -0,0 +1,382 @@
+    if 'remoted' in test_metadata and test_metadata['remoted']:
+        remoted_server = RemotedSimulator()
+        remoted_server.start()
+    else:
+        remoted_server = None
+    return remoted_server
+
+@pytest.mark.parametrize('test_configuration, test_metadata', zip(test_configuration, test_metadata), ids=test_cases_ids)
+def test_agentd_state(test_configuration, test_metadata, set_wazuh_configuration, remove_state_file, configure_local_internal_options, 
+                      truncate_monitored_files, clean_keys, add_keys, daemons_handler):
+    '''
+    description: Check that the statistics file 'wazuh-agentd.state' is created automatically
+                 and verify that the content of its fields is correct.
+
+    wazuh_min_version: 4.2.0
+
+    tier: 0
+
+    parameters:
+        - test_configuration:
+            type: data
+            brief: Configuration used in the test.
+        - test_metadata:
+            type: data
+            brief: Configuration cases.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - remove_state_file:
+            type: fixture
+            brief: Removes the wazuh-agentd.state file
+        - configure_local_internal_options:
+            type: fixture
+            brief: Set internal configuration for testing.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Reset the 'ossec.log' file and start a new monitor.
+        - clean_keys:
+            type: fixture
+            brief: Cleans keys file content
+        - add_keys:
+            type: fixture
+            brief: Adds keys to keys file
+        - daemons_handler:
+            type: fixture
+            brief: Handler of Wazuh daemons.
+
+    assertions:
+        - Verify that the 'wazuh-agentd.state' statistics file has been created.
+        - Verify that the information stored in the 'wazuh-agentd.state' statistics file
+          is consistent with the connection status to the 'wazuh-remoted' daemon.
+
+    input_description: An external YAML file (wazuh_conf.yaml) includes configuration settings for the agent.
+                       Different test cases that are contained in an external YAML file (wazuh_state_tests.yaml)
+                       that includes the parameters and their expected responses.
+
+    expected_output:
+        - r'pending'
+        - r'connected'
+    '''
+
+    # Start RemotedSimulator if test case need it
+    remoted_server = start_remoted_server(test_metadata) 
+
+    # Check fields for every expected output type
+    for expected_output in test_metadata['output']:
+        check_fields(expected_output, remoted_server)
+
+    #Shutdown simulator
+    if remoted_server:
+        remoted_server.destroy()
+    
+    
+def parse_state_file():
+    """Parse state file
+
+    Returns:
+        state info
+    """
+    # Wait until state file is dumped
+    wait_state_update()
```

Comment:
If you move the function to the framework, call this function before in the test.

Reply:
Addressed with https://github.com/wazuh/wazuh/pull/18353/commits/5a1b7778d254f00b85c507ace508492c18df7d0c